### PR TITLE
Readers details to support dynamic font size

### LINKED ISSUE #5521: [Mobile Payments] Support for Dynamic Type in reader connected screen
Outcome of #4348

Support for Dynamic Type in the the screen that shows the reader firmware version

| Text 135% | Text 310% |
| ---- | ---- |
|  |  |

pdfdoF-hD-p2

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #5521 

### Description

The PR replaces fixed size of the fonts and adjusts constraints so info about a connected reader readable with large font size

### Testing instructions

* Go to the settings -> IPP
* Connect to a reader
* Change the system font size to something big
* Notice that info about connected reader is readable now

### Screenshots

image

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderTableViewCells/ConnectedReaderTableViewCell.xib [hunk 1]
```diff
@@ -17,27 +18,21 @@
...
                         <nil key="textColor"/>
                         <nil key="highlightedColor"/>
                     </label>
-                    <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="44% Battery" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="rws-DM-4n4">
+                    <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" ambiguous="YES" text="44% Battery" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontForContentSizeCategory="YES" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="rws-DM-4n4">
                         <rect key="frame" x="20" y="34" width="374" height="15"/>
-                        <constraints>
-                            <constraint firstAttribute="height" constant="15" id="vhv-1f-sz5"/>
-                        </constraints>
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderTableViewCells/ConnectedReaderTableViewCell.xib:33]: Nice not having hard-coded height for these labels! 💯

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderTableViewCells/ConnectedReaderTableViewCell.xib [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="19455" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="20037" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderTableViewCells/ConnectedReaderTableViewCell.xib:2]: nit: in the Interface Builder, there seems to be some warnings about content size ambiguity that aren't in `trunk`. Is it possible to fix them?

**Review by jaclync — APPROVED** [review]
Nice work! Thanks for adding dynamic type support :shipit: 

One suggestion: maybe we can also make sure the card reader model isn't truncated? Setting the label's `numberOfLines` from 1 to 0 might help

`trunk` | PR branch
-- | --
Simulator Screen Shot - iPhone 11 - 2022-05-20 at 09 18 43 | Simulator Screen Shot - iPhone 11 - 2022-05-20 at 09 18 45

**Comment by kidinov** [comment]
@jaclync thank you for the review!

I set number of lines for battery and version labels to 0 b1436845cfb44aee6f629e821288f507ad98166a

But it keeps using as many lines as it wants =(. Maybe do you have an idea what's wrong?

image

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderTableViewCells/ConnectedReaderTableViewCell.xib (hunk 2 above)
Comment (kidinov) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderTableViewCells/ConnectedReaderTableViewCell.xib:2]: Good catch. Fixed it by clicking on some button which apparently automattically fixes that 
af5d5d60d1d84be3f9b0733b9d127821d05e291e

**Review by jaclync — APPROVED** [review]
> I set number of lines for battery and version labels to 0 https://github.com/woocommerce/woocommerce-ios/commit/b1436845cfb44aee6f629e821288f507ad98166a
> But it keeps using as many lines as it wants =(. Maybe do you have an idea what's wrong?

Setting `numberOfLines` to `0` does make a label take any number of lines, this is the expected behavior. Do you think this behavior isn't ideal? Alternatively, we could set the label to always be one line (default behavior `numberOfLines = 1`) and set `adjustsFontSizeToFitWidth = true` and `minimumScaleFactor` as in [this SO solution](https://stackoverflow.com/a/4865577/9185596). Personally, I think letting the label expand to multiple lines is more scalable because we don't have control over the length of the text (card reader model) and the app width could be narrow like in iPhone SE devices or split-screen mode. The text could still be truncated when the text still doesn't fit after `minimumScaleFactor` adjustment. It just so happened there is one character overflow in the screenshot 😆 

Changes LGTM 👍

**Comment by kidinov** [comment]
@jaclync I got you wrong at the beginning 🤦‍♂️ I thought that you want to make it fit on one line. I'm merging then!

**Pull request merged**